tests/test_callbacks.py

```diff
@@ -36,9 +41,59 @@ def test_callbacks():
     check_estimators_pickle("KerasClassifier", estimator)
     check_estimators_partial_fit_n_features("KerasClassifier", estimator)
     # Check that callback was called
-    estimator.fit([[0]], [1])  # quick fit
-    assert estimator.callbacks[0].called != SentinalCallback.called
+    estimator.fit([[0]], [1])
+    assert estimator.callbacks[0].train_counter == 1
     serialized_estimator = pickle.dumps(estimator)
     deserialized_estimator = pickle.loads(serialized_estimator)
-    assert deserialized_estimator.callbacks[0].called == estimator.callbacks[0].called
-    estimator.fit([[0]], [1])  # quick fit
+    assert deserialized_estimator.callbacks[0].train_counter == 1
+    deserialized_estimator.fit([[0]], [1])
+    assert deserialized_estimator.callbacks[0].train_counter == 2
+
+
+class Schedule:
+    """Simple exponential decay lr scheduler.
+    Modeled after https://keras.io/api/callbacks/learning_rate_scheduler/
+    Used to test nested object construction for callbacks.
+    """
+
+    def __init__(self, coef: float = 0.1) -> None:
+        self.coef = coef
+
+    def __call__(self, epoch: int, lr: float):
+        if epoch > 10:
+            lr = lr * math.exp(-self.coef)
+        return lr
+
+
+def test_callback_param_routing():
+    """Test that callbacks can be passed as routed parameters (eg. for grid search).
+
+    See https://github.com/adriangb/scikeras/issues/232
+    """
+    X, y = (
+        np.random.uniform(size=(100, 1)),
+        np.random.randint(low=0, high=2, size=(100,)),
+    )
+
+    def get_clf() -> keras.Model:
+        model = keras.models.Sequential()
+        model.add(keras.layers.InputLayer((1,)))
+        model.add(keras.layers.Dense(2, activation="softmax"))
+        return model
+
+    # using LRS with a nested instantiation of Schedule
+    clf = KerasClassifier(
+        get_clf,
+        epochs=15,
+        loss="sparse_categorical_crossentropy",
+        optimizer=keras.optimizers.SGD(
+            learning_rate=0.1
+        ),  # to mirror https://keras.io/api/callbacks/learning_rate_scheduler/
+        fit__callbacks=[keras.callbacks.LearningRateScheduler],
```

Comment:
Suggested change:

``` python
         fit__callbacks={"lr": keras.callbacks.LearningRateScheduler},
```

I think this will help with the ugly `0__0` syntax.

Reply:
Would this be in-lieu of, or in addition to?

Reply:
Incorporated, keeping compatibility with the `0__0` syntax as well.